src/dns/res.c

```diff
@@ -46,10 +47,26 @@ int get_resolv_dns(char *domain, size_t dsize, struct sa *nsv, uint32_t *n)
 	}
 
 	err = 0;
+#ifdef OPENBSD
 	for (i=0; i<min(*n, (uint32_t)state.nscount) && !err; i++) {
 		struct sockaddr_in *addr = &state.nsaddr_list[i];
 		err |= sa_set_sa(&nsv[i], (struct sockaddr *)addr);
 	}
+#else
+	union res_sockaddr_union *addr_union = mem_alloc(state.nscount * sizeof(union res_sockaddr_union), NULL);
+	int servers = res_getservers(&state, addr_union,  state.nscount);
```

Comment:
`res_getservers` does not exist on all platforms (use #if defined (DARWIN))

Reply:
That seems strange, since darwin has its own 'get_darwin_dns' in ns.c. This code gets called for #ifdef HAVE_RESOLV in ns.c.

Reply:
They are tried in prioritized order, only if HAVE_RESOLV is not available or `get_resolv_dns` errors, `get_darwin_dns` is called.
HAVE_RESOLV is used on linux too and works fine without a patch for IPv6. But `res_getservers` is BSD only.